### ISSUE TITLE
Weaviate allows to create a collection with unavailable modules

### ISSUE THREAD

**Issue by sebawita** [weaviate/weaviate#3111]
## TLDR
When I create a new collection, Weaviate allows me to add modules to the `moduleConfig`, which are not present.
This leads to unexpected behaviour at a later stage.

I would expect Weaviate to throw an error if I try to create a collection with modules that are not available.

## Long story
I have a Weaviate deployment with the following modules:
* `generative-openai`
* `text2vec-cohere`

The result from: `client.get_meta()`
```
{'hostname': 'http://[::]:8080',
 'modules': {'generative-openai': {'documentationHref': 'https://beta.openai.com/docs/api-reference/completions',
   'name': 'Generative Search - OpenAI'},
  'ref2vec-centroid': {},
  'text2vec-cohere': {'documentationHref': 'https://docs.cohere.ai/embedding-wiki/',
   'name': 'Cohere Module'}},
 'version': '1.19.6'}
```

I have a collection with the following `moduleConfig`:
```
    "moduleConfig": {
        "text2vec-cohere": {
            "model": "multilingual-22-12",
            "truncate": "RIGHT"
        },
        "generative-cohere": {
            "model": "command-xlarge-nightly",
        }
    },
```

Note: the moduleConfig refers to `generative-cohere`, which is not present in my modules.

When I run a query that contains `with_generate`, I get the following error:
```
'generate': {'error': 'OpenAI API Key: no api key found neither in request header: X-Openai-Api-Key nor in environment variable under OPENAI_APIKEY'
```

While I would expect to be notified, that the `generative-cohere` module is missing.

### Expected behaviour
When I create a new collection, Weaviate should check if all the provided modules exist.
I shouldn't be able to create a collection with `generative-cohere` if that module is not available.
